### PR TITLE
Sitepermissions location.

### PR DESCRIPTION
**Pull request by Amejia481**
@pocmo this is just a WIP to show you how DoorhangerPrompt and site permissions look like together :) 
[Demo]( https://drive.google.com/file/d/12hiVbcEkJzTw21H04_FXbyKFzMdQniTH/view?usp=sharing)

Most of the code is on `SitePermissionsFeature`, I had to include your unmerged branch to showcase it.

### CODE REVIEW

**Comment by pocmo** [comment]
@Amejia481 This looks pretty nice :)

**Comment by Amejia481** [comment]
@pocmo I updated the PR to only handle the location permission.
[Demo](https://drive.google.com/file/d/1X4uSTMeDo_cicR7vd7ljsoOBB16siHIR/view?usp=sharing)

I will wait for adding the changelog entry until we have handled more permissions to just have one entry.

File: components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsFeatureTest.kt [hunk 1]
```diff
@@ -62,68 +71,110 @@ class SitePermissionsFeatureTest {
...
 
-        assertFalse(session.contentPermissionRequest.isConsumed())
+        session.contentPermissionRequest = Consumable.from(permissionRequest)
 
-        sitePermissionFeature.onContentPermissionDeny("unknown_session", session.url)
+        val prompt = sitePermissionFeature.onContentPermissionRequested(session, permissionRequest)
 
-        assertFalse(session.contentPermissionRequest.isConsumed())
+        val negativeButton = prompt.buttons.find { !it.positive }
+        negativeButton?.onClick?.invoke()
```
Comment (pocmo) [inline components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsFeatureTest.kt]: nit: In tests when you expect something to exist then let's test explicitly: `negativeButton!!.onClick!!.invoke()`. If something in this chain does not exist it's good to fail early.

Comment (Amejia481) [reply components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsFeatureTest.kt]: 👍

**Pull request merged**